SIPs/sip_001.md

```diff
@@ -0,0 +1,172 @@
+    - An API user is not required to know about concurrent edition creation
+      processes running on a piece they try to create editions on
+      (meaning an API user can submit a request for creating editions, and even
+      though this process is still running, can send another request for
+      creating more editions)
+    - An API user must be able to create a distinct set of editions which
+      numbers do neither need to be in order, nor do they need to be consecutive
+      (All these examples shall be valid: [1, 2, 3], [3, 1, 2], [3, 1])
+    - An API user must be able to `freeze` or limit the unlimited editions of a
+      piece using a single HTTP request
+    - An API user must be able to track the creation of editions of a piece in a
+      modern-web kind of fashion (e.g. using websockets, instead of polling)
+
+
+## Problem
+
+Currently when creating editions, a user can make use of two endpoints:
+
+    1. Set `num_editions` when registering a piece `POST /api/pieces`
+    2. Set `num_editions` and `piece_id` using `POST /api/editions`
+
+When calling either of those endpoints, the user currently creates a set of
+limited editions in the range of 0 and the value of `num_editions` and a SPOOL
+ownership action (in form of a Bitcoin transaction) is triggered.
+
+For allowing unlimited editions to be implemented the following changes (that are
+not up for discussion in this SIP) will be made to the SPOOL protocol:
+
+    - When registering editions, a SPOOL `OwnershipEditions` Bitcoin transaction
+      will not be sent to the Bitcoin network immediately
+    - When doing actions(transfer, consign, loan) on a previously
+      registered/created edition, ownership actions on the Bitcoin network will
+      be recorded as usual
+    - When the user decides to limit his unlimited editions (we call this a
+      `freeze` action) a SPOOL `OwnershipEditions` Bitcoin transaction will be
+      sent to the Bitcoin network
+
+As this change needs to be implemented in version 0 of the API, its changes must
+not break the current endpoint's behavior. This exposes the following constraint:
+
+    - Using `num_editions` in `POST /api/pieces` and `POST /api/editions` still
+      needs to register a limited set of editions in a range of 0 to the value
+      of `num _editions`
+
+
+## Solution
+
+### Creation of unlimited editions
+
+Using `POST /api/pieces/1234/editions/` an API user can create a distinct set of unlimited
+editions using the following payload:
+
+```javascript
+POST /api/pieces/1234/editions/
+
+    {
+        "editions": [2, 1, 5, 59]
+    }
+```
+
+As mentioned in the requirements, here `editions` just specifies the
+edition numbers to be created. The numbers do not need to be ordered or
+consecutive.
+
+In response, the API returns a payload of the following form:
+
+
+```javascript
+202 Accepted
+
+    {
+        "notification": "We've succesfully received your request of registering
```

Comment:
I fully agree with this suggestion, but its actually out of the scope of this SIP.

If we'd like to change the notification construct (which is a rather heavy used one), we should draft its own SIP.